lib/measured/util.rb

```diff
@@ -0,0 +1,24 @@
+module Measured
+  class << self
+    alias_method :old_build, :build
+
+    def build(**kwargs, &block)
+      measurable = old_build(**kwargs, &block)
+      add_extensions(measurable)
+      measurable
+    end
+
+    private
+
+    def add_extensions(measurable)
+      measurable.units_with_aliases.each do |name|
+        Numeric.class_eval do
+          define_method(name) { measurable.new(self, name) }
+        end
+      end
+    end
+  end
+
+  add_extensions(Measured::Length) if defined?(Measured::Length)
+  add_extensions(Measured::Weight) if defined?(Measured::Weight)
```

Comment:
I'd really much prefer an implementation that didn't require referencing each of these subclasses manually.

Few reasons:
* Makes loading of these files order dependant
* Have to keep this file in sync with the base file where the length/width are required
* Another step for anyone defining a new unit system

I would much prefer an implementation that did this either when the unit system is defined, or by using subclass listing. Still, all of this has tricky ordering.